src/common/snippets/include/snippets/op/subgraph.hpp

```diff
@@ -143,28 +139,33 @@ class Subgraph : public ov::op::util::SubGraphOp {
     // Return estimated unique buffer count (upper bound). It's needed for tokenization
     static auto get_estimated_buffer_count(const ov::NodeVector& ops) -> size_t;
     static auto is_domain_sensitive_op(const std::shared_ptr<ov::Node>& op) -> bool;
+
+    void data_flow_shape_agnostic(const BlockedShapeVector& blocked_input_shapes = {},
+                                  const std::vector<ov::element::Type>& input_precisions = {},
+                                  const std::vector<ov::element::Type>& output_precisions = {},
+                                  const std::vector<snippets::pass::Manager::PositionedPass>& = {});
     std::shared_ptr<lowered::LinearIR>
-    convert_body_to_linear_ir(const std::shared_ptr<IShapeInferSnippetsFactory>& shape_infer_factory = std::make_shared<IShapeInferSnippetsFactory>()) const;
+    convert_body_to_linear_ir(const std::shared_ptr<IShapeInferSnippetsFactory>& shape_infer_factory = std::make_shared<IShapeInferSnippetsFactory>());
+    std::shared_ptr<Subgraph> clone() const;
 
 private:
-    void align_element_types(const BlockedShapeVector& outputShapes, const BlockedShapeVector& inputShapes);
-    void data_flow_transformations(const std::vector<snippets::pass::Manager::PositionedPass>& backend_passes);
     void control_flow_transformations(lowered::LinearIR& linear_ir,
+                                      LoweringResult& lowering_result,
                                       const lowered::pass::PassPipeline& backend_passes_pre_common,
-                                      const lowered::pass::PassPipeline& backend_passes_post_common);
+                                      const lowered::pass::PassPipeline& backend_passes_post_common) const;
     void init_config();
     // Count of Subgraph virtual ports:
     //  - Potential non-scalar Constants that will be created after some transformations (At the moment it's relevant only for FakeQuantize decomposition)
     // NOTE: To avoid overheads in each calculation of this count (for example, in validate_and_type_infer()),
     //       we should MANUALLY calculate it where it needed.
     size_t m_virtual_port_count = 0;
-    size_t m_buffer_scratchpad = 0lu;
     Shape exec_domain = {};
     std::shared_ptr<ov::snippets::Generator> m_generator = nullptr;
+    // true if Subgraph was created with dynamic shapes. Allows skip shape inference
+    bool m_is_dynamic = false;
```

Comment:
The idea was to remember this state, since we know that this property can't change after the creation of subgraph, while `is_dynamic()` checks all inputs on every call. On the other hand, we call it only from `get_master_shape()`, so I think it's all right to use `is_dynamic()` after all.